File: server/src/game/conf/RoleInitConfig.go
package conf

import (
	//"bufio"
	"encoding/json"
	"fmt"
	//"io"
	"io/ioutil"
	"os"
	//"strings"
)

type RoleInitConfigElem struct {
	Id int
	IconID int
	Gold int
	Diamond int
	Body int
	Job string
	Name string
	InitSuit int
	Cloth []int
	Mood int
	Title int
	Exp int
	Mail int
	House int
	Car int
	Ship int
	ItemIds []int
	
}

type RoleInitConfig0 struct {
	Dic map[int]*RoleInitConfigElem
	Arr []*RoleInitConfigElem
}

var (
	RoleInitConfig RoleInitConfig0
)

func loadRoleInitConfig(path string) {
	file, err := os.Open(path)
	defer func() {
		if file != nil {
			file.Close()
		}
	}()
	if err != nil {
		panic(fmt.Sprint("load config error, path=", path))
	}

	RoleInitConfig.Dic = make(map[int]*RoleInitConfigElem)
	RoleInitConfig.Arr = make([]*RoleInitConfigElem, 0, 32)

	bt,err  :=  ioutil.ReadAll(file)
	json.Unmarshal(bt, &RoleInitConfig.Arr)
	for _,conf := range RoleInitConfig.Arr{
		RoleInitConfig.Dic[conf.Id] = conf
	}
	
}

File: tools/conf/output/ts/RoleInitConfig.ts
class RoleInitConfig {
    private static _dic:Dictionary<RoleInitConfig> = new Dictionary<RoleInitConfig>();
    private static _arr:Array<RoleInitConfig> = new Array<RoleInitConfig>();

	 public Id :number ;
     public IconID :number ;
     public Gold :number ;
     public Diamond :number ;
     public Body :number ;
     public Job :string ;
     public Name :string ;
     public InitSuit :number ;
     public Cloth : Array<number>  =new Array<number>();
     public Mood :number ;
     public Title :number ;
     public Exp :number ;
     public Mail :number ;
     public House :number ;
     public Car :number ;
     public Ship :number ;
     public ItemIds : Array<number>  =new Array<number>();
    

    
    public constructor(){

    }


    public static get Dic():Dictionary<RoleInitConfig>{
        return  RoleInitConfig._dic;
    }
    public static get Arr():Array<RoleInitConfig>{
        return RoleInitConfig._arr;
    }
    public static parse(json:any[]){

        for(var i = 0; i < json.length; i++){
            let config:RoleInitConfig = <RoleInitConfig>json[i];
            RoleInitConfig._arr.push(config);
            RoleInitConfig._dic.add(config.Id,config);
        }
    }
}